Jenkinsfile

```diff
@@ -3,79 +3,143 @@
-            junit 'acceptance-tests/target/surefire-reports/*.xml'
-          }
-
-          stage('ATH - Jenkins 2.32.3') {
-            sh "cd acceptance-tests && ./run.sh -v=2.32.3 --no-selenium --settings='-s ${env.WORKSPACE}/settings.xml'"
-            junit 'acceptance-tests/target/surefire-reports/*.xml'
-          }
-
-          stage('ATH - Jenkins 2.19.4') {
-            sh "cd acceptance-tests && ./run.sh -v=2.19.4 --no-selenium --settings='-s ${env.WORKSPACE}/settings.xml'"
-            junit 'acceptance-tests/target/surefire-reports/*.xml'
-          }
-        }
+try {
+  stage("Build and test") {
+    parallel( 
+        "Build and Unit" : {
+          buildAndTest()
+        },
+        "ATH" : {
+          acceptanceTests()
+        }            
+    )    
+  }  
+} finally {
+  stage("Notifying") {
+      notifications()
+  }
+  
+}
 
+def buildAndTest() {
+  node {    
+    prepareEnvironment()    
+    docker.image('blueocean_build_env').inside {
+      withEnv(['GIT_COMMITTER_EMAIL=me@hatescake.com','GIT_COMMITTER_NAME=Hates','GIT_AUTHOR_NAME=Cake','GIT_AUTHOR_EMAIL=hates@cake.com']) {
+        try {
+          fullBuildSteps()                  
+        } catch(err) {
+          handleBuildFailure(err)
+        } finally {                      
+          deleteDir()          
+        }
+      }
+    }
+  }
+}
 
-      } catch(err) {
-        currentBuild.result = "FAILURE"
+def acceptanceTests() {
+  node {    
+    prepareEnvironment()
+    sh "./acceptance-tests/runner/scripts/start-selenium.sh"    
 
-        if (err.toString().contains('exit code 143')) {
-          currentBuild.result = "ABORTED"
-        }
-      } finally {
-        stage('Cleanup') {
+    docker.image('blueocean_build_env').inside("--net=container:blueo-selenium") {
+      withEnv(['GIT_COMMITTER_EMAIL=me@hatescake.com','GIT_COMMITTER_NAME=Hates','GIT_AUTHOR_NAME=Cake','GIT_AUTHOR_EMAIL=hates@cake.com']) {
+        try {
+          sh "mvn install -DskipTests"
+          runAcceptanceTests()
+          runWeeklyPermutationATH()
+        } catch(err) {
+          handleBuildFailure(err)
+        } finally {
           sh "${env.WORKSPACE}/acceptance-tests/runner/scripts/stop-selenium.sh"
-          sendhipchat()
-          deleteDir()
+          deleteDir()          
         }
       }
     }
   }
 }
 
 
-def sendhipchat() {
+
+/** 
+* The build steps for a build and unit test
+* Includes some npm sanity checks. 
+*/
+def fullBuildSteps() {
+  sh 'npm --prefix ./blueocean-core-js install'
+  sh 'npm --prefix ./blueocean-core-js run gulp'
+  sh "mvn install -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -Dmaven.test.failure.ignore -s settings.xml"
+  junit '**/target/surefire-reports/TEST-*.xml'
+  junit '**/reports/junit.xml'
+  archive '*/target/*.hpi'
+  
+  sh "node ./bin/checkdeps.js"
+  sh "node ./bin/checkshrinkwrap.js"
```

Comment:
it needs to have a mvn install run anyway before they can run ... so it wouldn't really save much time. We could split up the mvn runs in to build vs test, but net result would be to slow down (but get slightly earlier feedback)